src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Shadow/ShadowEffect.shared.cs

```diff
@@ -0,0 +1,106 @@
+
+		internal const string OffsetXPropertyName = "OffsetX";
+
+		internal const string OffsetYPropertyName = "OffsetY";
+
+		public static readonly BindableProperty ColorProperty = BindableProperty.CreateAttached(
+			ColorPropertyName,
+			typeof(Color),
+			typeof(ShadowEffect),
+			Color.Default,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty	OpacityProperty = BindableProperty.CreateAttached(
+			OpacityPropertyName,
+			typeof(double),
+			typeof(ShadowEffect),
+			-1.0,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty RadiusProperty = BindableProperty.CreateAttached(
+			RadiusPropertyName,
+			typeof(double),
+			typeof(ShadowEffect),
+			-1.0,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty OffsetXProperty = BindableProperty.CreateAttached(
+			OffsetXPropertyName,
+			typeof(double),
+			typeof(ShadowEffect),
+			.0,
+			propertyChanged: TryGenerateEffect);
+
+		public static readonly BindableProperty OffsetYProperty = BindableProperty.CreateAttached(
+			OffsetYPropertyName,
+			typeof(double),
+			typeof(ShadowEffect),
+			.0,
+			propertyChanged: TryGenerateEffect);
+
+		public ShadowEffect()
+			: base(EffectIds.ShadowEffect)
+		{
+#if __ANDROID__
+			if (System.DateTime.Now.Ticks < 0)
+				_ = new Xamarin.CommunityToolkit.Android.Effects.PlatformShadowEffect();
+#elif __IOS__
+			if (System.DateTime.Now.Ticks < 0)
+				_ = new Xamarin.CommunityToolkit.iOS.Effects.PlatformShadowEffect();
+#elif __MACOS__
+			if (System.DateTime.Now.Ticks < 0)
+				_ = new Xamarin.CommunityToolkit.macOS.Effects.PlatformShadowEffect();
+#endif
+		}
+
+		public static Color GetColor(BindableObject bindable)
+			=> (Color)bindable.GetValue(ColorProperty);
+
+		public static void SetColor(BindableObject bindable, Color value)
+			=> bindable.SetValue(ColorProperty, value);
+
+		public static double GetOpacity(BindableObject bindable)
+			=> (double)bindable.GetValue(OpacityProperty);
+
+		public static void SetOpacity(BindableObject bindable, double value)
+			=> bindable.SetValue(OpacityProperty, value);
+
+		public static double GetRadius(BindableObject bindable)
+			=> (double)bindable.GetValue(RadiusProperty);
+
+		public static void SetRadius(BindableObject bindable, double value)
+			=> bindable.SetValue(RadiusProperty, value);
+
+		public static double GetOffsetX(BindableObject bindable)
+			=> (double)bindable.GetValue(OffsetXProperty);
+
+		public static void SetOffsetX(BindableObject bindable, double value)
+			=> bindable.SetValue(OffsetXProperty, value);
+
+		public static double GetOffsetY(BindableObject bindable)
+			=> (double)bindable.GetValue(OffsetYProperty);
+
+		public static void SetOffsetY(BindableObject bindable, double value)
+			=> bindable.SetValue(OffsetYProperty, value);
+
+		static void TryGenerateEffect(BindableObject bindable, object oldValue, object newValue)
+		{
```

Comment:
Should we check and remove the oldValue? As we did [here](https://github.com/xamarin/XamarinCommunityToolkit/blob/00dfb4d6526dd7494d3a8fb273533438a89d2249/src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/IconTintColor/IconTintColorEffect.shared.cs#L18)

Reply:
Hmmm, we can do it, if you prefer (but now if Default color selected we just ignore other properties)

Reply:
Implemented